eth/block_specifier.go

```diff
@@ -0,0 +1,137 @@
+package eth
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+type BlockSpecifier struct {
+	Number           *Quantity
+	Tag              *Tag
+	Hash             *Hash
+	RequireCanonical *bool
+	Raw              bool
+}
+
+func NewBlockSpecifierFromString(value string) (*BlockSpecifier, error) {
+	dec, err := NewBlockNumberOrTag(value)
```

Comment:
minor: while outside the spec, geth also supports converting a hash string into a block specifier.  If we wanted to add support for this, I think it'd just be a matter of trying `eth.NewHash(value)` for a block number, and falling through to this line if it returns an error.

Reply:
Added in [this commit](https://github.com/INFURA/go-ethlibs/pull/25/commits/4a945f438eb28b7e924be629714843cb7b47d55f)